New type guard checks have the wrong input type

the type guard check isArrayOfNodes is listed as taking `SelectedValue` which is a problem since that type no longer includes an array of Nodes.

Expected definition:
```ts
export function isArrayOfNodes(value: SelectReturnType): value is Node[];
```

Ah yes, same problem :(

```
node_modules/xpath/xpath.d.ts:42:64 - error TS2677: A type predicate's type must be assignable to its parameter's type.
  Type 'Node[]' is not assignable to type 'SelectedValue'.

42 export function isArrayOfNodes(value: SelectedValue): value is Node[];
                                                                  ~~~~~~


Found 1 error in node_modules/xpath/xpath.d.ts:42
```

--- a/xpath.d.ts
+++ b/xpath.d.ts
@@ -39,7 +39,7 @@ export function useNamespaces(namespaceMap: Record<string, string>): XPathSelect
 
 // Type guards to narrow down the type of the selected type of a returned Node object
 export function isNodeLike(value: SelectedValue): value is Node;
-export function isArrayOfNodes(value: SelectedValue): value is Node[];
+export function isArrayOfNodes(value: SelectReturnType): value is Node[];
 export function isElement(value: SelectedValue): value is Element;
 export function isAttribute(value: SelectedValue): value is Attr;
 export function isTextNode(value: SelectedValue): value is Text;

